### PR TITLE
docs: Rename map_to_app, add relay container docs #trivial

### PR DESCRIPTION
**Pull request by brainbicycle**
Co-authored-by: Mounir Dhahri <mounir.dhahri@ensi-uma.tn>

The type of this PR is: **Documentation**

### Description

- Renames map_to_app to preferred_practices 
- Adds section in preferred_practices on relay containers

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))

[MX-434]: https://artsyproduct.atlassian.net/browse/MX-434

### CODE REVIEW

File: docs/preferred_practices.md [hunk 1]
```diff
@@ -71,6 +71,12 @@ Data should be loaded from [Metaphysics](https://github.com/artsy/metaphysics),
   - [A top-level Relay component](https://github.com/artsy/eigen/blob/39644610eb2a5609d992f434a7b37b46e0953ff4/src/lib/Scenes/Collection/Collection.tsx)
   - [A fragment container](https://github.com/artsy/eigen/blob/39644610eb2a5609d992f434a7b37b46e0953ff4/src/lib/Scenes/Collection/Components/FeaturedArtists.tsx)
 
+### Prefer Relay containers (Higher Order Components) over Hooks
+
+We have a preference for Relay containers due to `relay-hooks` hooks not being compatible with Relay containers which represent the majority of our components using Relay.
```
Comment (damassi) [inline docs/preferred_practices.md:76]: _the greater good_  👍

**Review by ashfurrow — APPROVED** [review]
Thanks Brian and Mounir!

**Comment by ashfurrow** [comment]
#squashongreen

**Comment by ashfurrow** [comment]
Hmm! I expected a label. Let's try #mergeongreen ?

**Comment by ashfurrow** [comment]
Sorry for the noise, testing Peril.

One more time: #squashongreen ?

**Comment by ashfurrow** [comment]
Okay, looks like the broken part is only the marking as Squash-on-Green with a GitHub label with #squashongreen. I'll look into it more on Monday /cc @admbtlr

**Pull request merged**